### PR TITLE
Added calendar functionality, works for arbitrary month and year inputs

### PR DESCRIPTION
**Pull request by ShreyansK2000**
You may want to peek under the hood and see if there are memory/speed optimizations you think should be made. If so, let me know what I can change.

### CODE REVIEW

File: src/plugins/cal.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                var monthName = monthNames[parseInt(parse[1]) - 1];
+                var date = new Date (year, parseInt(parse[1]) - 1); 
+            } else {
+                var date = new Date();
+                var monthName = monthNames[date.getMonth()];
+                var year = date.getFullYear();
+                warning = "Printing for current month and year" + "\n";
+            }
+
+            (warning) ? this.parent.terminal.writeln(warning + "\n") : this.parent.terminal.writeln("\n");
```
Comment (lcfyi) [inline src/plugins/cal.js]: Does this do anything? It seems like testing the truthy value of `""` isn't necessary when you already assign it to an empty string in the beginning.

File: src/plugins/cal.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            
+            let monthLine = "         " + monthName + " " + year + "         ";
+            this.parent.terminal.writeln(monthLine);
+            this.parent.terminal.writeln(dayLine);
+            this.parent.terminal.writeln(generateCalString(date));
+        } catch(error) {
+            var val = "cal: " + error.message;
```
Comment (lcfyi) [inline src/plugins/cal.js:94]: Not a big deal, but was there any reason why you chose to use a `var` here?

File: src/plugins/cal.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+function generateCalString(date) {
+    let startOfMonth = new Date(date.getFullYear(), date.getMonth(), 1).getDay();
+    let numDays =  32 - new Date(date.getFullYear(), date.getMonth(), 32).getDate();
+    let daysIndex = startOfMonth + numDays;
+    var stringOut = "";
+
+    var i;
+    for (i = 0; i < daysIndex; i++) {
+        if (i >= startOfMonth) {
+            var date = i - startOfMonth + 1;
```
Comment (lcfyi) [inline src/plugins/cal.js]: Similar to above, it doesn't seem like you're benefitting from using `var` here.

File: src/plugins/cal.js (hunk 1 above)
Comment (ShreyansK2000) [reply src/plugins/cal.js]: I believe "" returns false, so I check here if there is anything in the warning string afterall

File: src/plugins/cal.js (hunk 2 above)
Comment (ShreyansK2000) [reply src/plugins/cal.js:94]: No reason for using val, this part was copy pasted from some other plugin and I left it almost as it is

File: src/plugins/cal.js (hunk 3 above)
Comment (ShreyansK2000) [reply src/plugins/cal.js]: No real reason for var again, can change it

**Comment by lcfyi** [comment]
Same as [this](https://github.com/lcfyi/decabyt.es/pull/8#issuecomment-549068140). 

Discard your changes to termprops.js and rebase from master.

**Pull request closed without merging**